src/tracker.jl

```diff
@@ -0,0 +1,329 @@
+    else
+        return lower(C.factors)
+    end
+end
+
+LinearAlgebra.LowerTriangular(A::TrackedMatrix) = lower(A)
+lower(A::TrackedMatrix) = track(lower, A)
+@grad lower(A) = lower(Tracker.data(A)), ∇ -> (lower(∇),)
+
+LinearAlgebra.UpperTriangular(A::TrackedMatrix) = upper(A)
+upper(A::TrackedMatrix) = track(upper, A)
+@grad upper(A) = upper(Tracker.data(A)), ∇ -> (upper(∇),)
+
+function LinearAlgebra.cholesky(A::TrackedMatrix; check=true)
+    factors_info = turing_chol(A, check)
+    factors = factors_info[1]
+    info = data(factors_info[2])
+    return Cholesky{eltype(factors), typeof(factors)}(factors, 'U', info)
+end
+function LinearAlgebra.cholesky(A::Symmetric{<:Any, <:TrackedMatrix}; check=true)
+    uplo = A.uplo == 'U' ? (:U) : (:L)
+    factors_info = symm_turing_chol(parent(A), check, uplo)
+    factors = factors_info[1]
+    info = data(factors_info[2])
+    return Cholesky{eltype(factors), typeof(factors)}(factors, 'U', info)
+end
+
+turing_chol(A::TrackedMatrix, check) = track(turing_chol, A, check)
+@grad function turing_chol(A::AbstractMatrix, check)
+    C, back = ZygoteRules.pullback(data(A), check) do A, check
+        cholesky(A, check=check)
+    end
+    return (C.factors, C.info), Δ->back((factors=data(Δ[1]),))
+end
+
+symm_turing_chol(A::TrackedMatrix, check, uplo) = track(symm_turing_chol, A, check, uplo)
+@grad function symm_turing_chol(A::AbstractMatrix, check, uplo)
+    C, back = ZygoteRules.pullback(data(A), check, uplo) do A, check, uplo
+        cholesky(Symmetric(A, uplo), check=check)
+    end
+    return (C.factors, C.info), Δ->back((factors=data(Δ[1]),))
+end
+
+# Specialised logdet for cholesky to target the triangle directly.
+logdet_chol_tri(U::AbstractMatrix) = 2 * sum(log, U[diagind(U)])
+logdet_chol_tri(U::TrackedMatrix) = track(logdet_chol_tri, U)
+@grad function logdet_chol_tri(U::AbstractMatrix)
+    U_data = data(U)
+    return logdet_chol_tri(U_data), Δ->(Matrix(Diagonal(2 .* Δ ./ diag(U_data))),)
+end
+
+function LinearAlgebra.logdet(C::Cholesky{<:TrackedReal, <:TrackedMatrix})
+    return logdet_chol_tri(C.U)
+end
+
+function zygote_ldiv(A::TrackedMatrix, B::TrackedVecOrMat)
+    return track(zygote_ldiv, A, B)
+end
+function zygote_ldiv(A::TrackedMatrix, B::AbstractVecOrMat)
+    return track(zygote_ldiv, A, B)
+end
+zygote_ldiv(A::AbstractMatrix, B::TrackedVecOrMat) =  track(zygote_ldiv, A, B)
+@grad function zygote_ldiv(A, B)
+    Y, back = ZygoteRules.pullback(\, data(A), data(B))
+    return Y, Δ->back(data(Δ))
+end
+
+function Base.:\(a::Cholesky{<:TrackedReal, <:TrackedArray}, b::AbstractVecOrMat)
+    return (a.U \ (a.U' \ b))
+end
+
+## SpecialFunctions ##
+
+SpecialFunctions.logabsgamma(x::TrackedReal) = track(logabsgamma, x)
+@grad function SpecialFunctions.logabsgamma(x::Real)
+    return logabsgamma(data(x)), Δ -> (digamma(data(x)) * Δ[1],)
+end
+ZygoteRules.@adjoint function SpecialFunctions.logabsgamma(x::Real)
+    return logabsgamma(x), Δ -> (digamma(x) * Δ[1],)
+end
```

Comment:
This is definitely not needed anymore - there exist adjoints for `logabsgamma` in ChainRules (should be moved to SpecialFunctions at some point, but that's just for now) and Zygote supports ChainRules, so there should be no need for a dedicated adjoint for Zygote.

Reply:
```suggestion
```